include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,793 @@
+#define FDCAN_IE_PEDE					(1 << 22)
+#define FDCAN_IE_ARAE					(1 << 23)
+/**@}*/
+
+/** @defgroup fdcan_ils FDCAN_ILS interrupt line select flags
+ * @{
+ */
+#define FDCAN_ILS_RxFIFO0				(1 << 0)
+#define FDCAN_ILS_RxFIFO1				(1 << 1)
+#define FDCAN_ILS_SMSG					(1 << 2)
+#define FDCAN_ILS_TFERR					(1 << 3)
+#define FDCAN_ILS_MISC					(1 << 4)
+#define FDCAN_ILS_BERR					(1 << 5)
+#define FDCAN_ILS_PERR					(1 << 6)
+/**@}*/
+
+#define FDCAN_ILE_INT0					(1 << 0)
+#define FDCAN_ILE_INT1					(1 << 1)
+
+#define FDCAN_RXGFC_RRFE				(1 << 0)
+#define FDCAN_RXGFC_RRFS				(1 << 1)
+/* ANFE[1:0]: Accept non-matching frames w/ extended ID */
+#define FDCAN_RXGFC_ANFE_SHIFT			2
+#define FDCAN_RXGFC_ANFE_MASK			(0x3 << FDCAN_RXGFC_ANFE_SHIFT)
+
+/* ANFS[1:0]: Accept non-matching frames w/ standard ID */
+#define FDCAN_RXGFC_ANFS_SHIFT			4
+#define FDCAN_RXGFC_ANFS_MASK			(0x3 << FDCAN_RXGFC_ANFS_SHIFT)
+
+#define FDCAN_RXGFC_F1OM				(1 << 8)
+#define FDCAN_RXGFC_F0OM				(1 << 9)
+/* LSS[4:0]: List size of standard ID filters */
+#define FDCAN_RXGFC_LSS_SHIFT			16
+#define FDCAN_RXGFC_LSS_MASK			(0x1F << FDCAN_RXGFC_LSS_SHIFT)
+
+/* LSE[3:0]: List size of extended ID filters */
+#define FDCAN_RXGFC_LSE_SHIFT			24
+#define FDCAN_RXGFC_LSE_MASK			(0xF << FDCAN_RXGFC_LSE_SHIFT)
+
+
+/* EIDM[28:0]: Extended ID mask for filtering */
+#define FDCAN_XIDAM_EIDM_SHIFT			0
+#define FDCAN_XIDAM_EIDM_MASK			(0x1FFFFFFF << FDCAN_XIDAM_EIDM_SHIFT)
+
+
+/* BIDX[2:0]: Buffer index */
+#define FDCAN_HPMS_BIDX_SHIFT			0
+#define FDCAN_HPMS_BIDX_MASK			(0x7 << FDCAN_HPMS_BIDX_SHIFT)
+
+/* MSI[1:0]: Message storage indicator */
+#define FDCAN_HPMS_MSI_SHIFT			6
+#define FDCAN_HPMS_MSI_MASK				(0x3 << FDCAN_HPMS_MSI_SHIFT)
+
+/* FIDX[4:0]: Filter index */
+#define FDCAN_HPMS_FIDX_SHIFT			8
+#define FDCAN_HPMS_FIDX_MASK			(0x1F << FDCAN_HPMS_FIDX_SHIFT)
+
+#define FDCAN_HPMS_FLS					(1 << 15)
+
+/* F0FL[3:0]: Fill level of Rx FIFO 0 */
+#define FDCAN_RXF0S_F0FL_SHIFT			0
+#define FDCAN_RXF0S_F0FL_MASK			(0xF << FDCAN_RXF0S_F0FL_SHIFT)
+
+/* F0GI[1:0]: Get index of Rx FIFO 0 */
+#define FDCAN_RXF0S_F0GI_SHIFT			8
+#define FDCAN_RXF0S_F0GI_MASK			(0x3 << FDCAN_RXF0S_F0GI_SHIFT)
+
+/* F0PI[1:0]: Put index of Rx FIFO 0 */
+#define FDCAN_RXF0S_F0PI_SHIFT			16
+#define FDCAN_RXF0S_F0PI_MASK			(0x3 << FDCAN_RXF0S_F0PI_SHIFT)
+
+#define FDCAN_RXF0S_F0F					(1 << 24)
+#define FDCAN_RXF0S_RF0L				(1 << 25)
+
+/* R0AI[2:0]: Rx FIFO 0 acknowledge index */
+#define FDCAN_RXF0A_R0AI_SHIFT			0
+#define FDCAN_RXF0A_R0AI_MASK			(0x7 << FDCAN_RXF0A_R0AI_SHIFT)
+
+
+/* F1FL[3:0]: Fill level of Rx FIFO 1 */
+#define FDCAN_RXF1S_F1FL_SHIFT			0
+#define FDCAN_RXF1S_F1FL_MASK			(0xF << FDCAN_RXF1S_F1FL_SHIFT)
+
+/* F1GI[1:0]: Get index of Rx FIFO 1 */
+#define FDCAN_RXF1S_F1GI_SHIFT			8
+#define FDCAN_RXF1S_F1GI_MASK			(0x3 << FDCAN_RXF1S_F1GI_SHIFT)
+
+/* F1PI[1:0]: Put index of Rx FIFO 1 */
+#define FDCAN_RXF1S_F1PI_SHIFT			16
+#define FDCAN_RXF1S_F1PI_MASK			(0x3 << FDCAN_RXF1S_F1PI_SHIFT)
```

Comment:
it might be possible tosimpliy this series, as it's all, 0xf, 0x3@8, 0x3@16, for both fifos? probably no gain rightnow, but might make some stuff easier.

Reply:
I'd like to keep naming consistent, so if FDCAN_<register>_<parameter> is defined somewhere, it should also be defined for all other valid combinations of register and parameter even if those are duplications. But yes, actual values could be put into "register agnostic" macro and then just referenced here.